gandalf.js

```diff
@@ -1,6 +1,7 @@
 /**
```

Comment:
SCRIPT-This logic might be clearer if you used a switch statement instead of multiple if-else statements.